### PR TITLE
SECURITY: prevent DeleteFilePost doing arbitrary deletion

### PR DESCRIPTION
**Pull request by zeripath**
Unfortunately a suitably malformed request to DeleteFilePost will allow arbitrary deletion. Further it was also possible to adjust the .git directories on editFilePost and UploadFilePost.

### CODE REVIEW

**Comment by jonasfranz** [comment]
@go-gitea/owners I think we have to backport this to 1.6 since this is quite critical IMHO.

File: routers/repo/editor.go [hunk 1]
```diff
@@ -163,7 +163,10 @@ func editFilePost(ctx *context.Context, form auth.EditRepoFileForm, isNewFile bo
 		branchName = form.NewBranchName
 	}
 
-	form.TreePath = strings.Trim(path.Clean("/"+form.TreePath), " /")
+	form.TreePath = cleanUploadFileName(form.TreePath)
+	if len(form.TreePath) == 0 {
+		ctx.Error(500, "Upload file name is invalid")
```
Comment (jonasfranz) [inline routers/repo/editor.go:168]: missing return?

**Review by jonasfranz — CHANGES_REQUESTED** [review]

File: routers/repo/editor.go (hunk 1 above)
Comment (zeripath) [reply routers/repo/editor.go:168]: Oops. Repushed.

File: routers/repo/editor.go [hunk 2]
```diff
@@ -373,6 +377,13 @@ func DeleteFile(ctx *context.Context) {
 func DeleteFilePost(ctx *context.Context, form auth.DeleteRepoFileForm) {
 	ctx.Data["PageIsDelete"] = true
 	ctx.Data["BranchLink"] = ctx.Repo.RepoLink + "/src/" + ctx.Repo.BranchNameSubURL()
+
+	ctx.Repo.TreePath = cleanUploadFileName(ctx.Repo.TreePath)
```
Comment (jonasfranz) [inline routers/repo/editor.go]: Is this just a check or is this `TreePath`actually used anywhere else?

**Review by jonasfranz — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

**Comment by techknowlogick** [comment]
@zeripath thanks for PR :) as @jonasfranz has said, please backport to both release/v1.7 and release/v1.6

**Pull request merged**